examples/car_fsm.erl

```diff
@@ -0,0 +1,543 @@
+%% `proper:quickcheck(car_fsm:prop_targeted_distance(), 1000).'
+prop_distance_targeted() ->
+  ?FORALL_TARGETED(
+     Cmds, more_commands(3, proper_fsm:targeted_commands(?MODULE)),
+     ?TRAPEXIT(
+        begin
+          start_link(),
+          {_H, {_, S}, R} = proper_fsm:run_commands(?MODULE, Cmds),
+          stop(),
+          #state{distance = Distance, burnt = Burnt} = S,
+          Consumption = case Distance > 0 of
+                          true -> 100 * Burnt / Distance;
+                          false -> 0
+                        end,
+          UV = Distance,
+          ?MAXIMIZE(UV),
+          ?WHENFAIL(
+             io:format("Distance: ~p~nConsumption: ~p~n",
+                       [Distance, Consumption]),
+             aggregate(command_names(Cmds),
+                       R =:= ok andalso (Distance < ?DISTANCE orelse
+                                         Consumption > ?CONSUMPTION)))
+        end)).
+
+%% This should fail most of the times with:<br/>
+%% `proper:quickcheck(car_fsm:prop_targeted_distance_init(), 1000).'
+prop_distance_targeted_init() ->
+  State = {initial_state(), initial_state_data()},
+  ?FORALL_TARGETED(
+     Cmds, more_commands(3, proper_fsm:targeted_commands(?MODULE, State)),
+     ?TRAPEXIT(
+        begin
+          start_link(),
+          {_H, {_, S}, R} = proper_fsm:run_commands(?MODULE, Cmds),
+          stop(),
+          #state{distance = Distance, burnt = Burnt} = S,
+          Consumption = case Distance > 0 of
+                          true -> 100 * Burnt / Distance;
+                          false -> 0
+                        end,
+          UV = Distance,
+          ?MAXIMIZE(UV),
+          ?WHENFAIL(
+             io:format("Distance: ~p~nConsumption: ~p~n",
+                       [Distance, Consumption]),
+             aggregate(command_names(Cmds),
+                       R =:= ok andalso (Distance < ?DISTANCE orelse
+                                         Consumption > ?CONSUMPTION)))
+        end)).
+
+%% -----------------------------------------------------------------------------
+%% Calculation Functions
+%% -----------------------------------------------------------------------------
+
+
+travel_calculations(Distance, Speed, Fuel) when Speed > 0 ->
+  Consumption = fuel_consumption(Speed),
+  Burn = Consumption * Distance / 100,
+  case Burn > Fuel of
+    true -> {Fuel * 100 / Consumption, Fuel};
+    false -> {Distance, Burn}
+  end;
+travel_calculations(_D, _S, _F) ->
+  {0, 0.0}.
+
+acceleration_calculations({Speed, Accel}, Fuel) when Accel > 0 ->
+  Acceleration = case Speed + Accel > ?MAX_SPEED of
+                   true -> ?MAX_SPEED - Speed;
+                   false -> Accel
+                 end,
+  Consumption = fuel_consumption(Speed, Acceleration),
+  Distance = calculate_distance(Speed, Acceleration),
+  Burn = Consumption * Distance / 100,
+  case Burn > Fuel of
+    true -> acceleration_calculations({Speed, Acceleration - ?ACCELERATION},
```

Comment:
I find this two line combination to be of terrible style!
Looks _much_ better as
```erlang
  true ->
    acceleration_calculations(...)
```